src/EA.Weee.Database/scripts/Update/Release 2/Sprint12/20190704-1422-create-spgAatfAeReturnDataCsvData.sql

```diff
@@ -0,0 +1,163 @@
+-- Description:	This stored procedure is used to provide the data for the admin report of aatf/ae
+--				that have/haven't submitted a data return within
+--				the limits of the specified parameters.
+-- =============================================
+CREATE PROCEDURE [AATF].[spgAatfAeReturnDataCsvData]
+	@ComplianceYear INT,
+	@Quarter INT,
+	@FacilityType INT,
+	@ReturnStatus INT,
+	@CA UNIQUEIDENTIFIER,
+	@Area UNIQUEIDENTIFIER,
+	@PanArea UNIQUEIDENTIFIER
+AS
+BEGIN
+
+SET NOCOUNT ON;
+
+DECLARE @AATF TABLE
+(
+	AatfId					UNIQUEIDENTIFIER NOT NULL,
+	Name					NVARCHAR(256) NOT NULL,
+	ApprovalNumber			NVARCHAR(20) NOT NULL,	
+	OrganisationName		NVARCHAR(256) NULL,
+	CompetentAuthorityAbbr	NVARCHAR(65) NOT NULL,
+	CompetentAuthorityId	UNIQUEIDENTIFIER NOT NULL
+)
+
+DECLARE @RETURN TABLE
+(
+	AatfId					UNIQUEIDENTIFIER NOT NULL,
+	ReturnId				UNIQUEIDENTIFIER NOT NULL,
+	ReturnStatus			INT NOT NULL,
+	CreatedDate				DATETIME NOT NULL,
+	SubmittedDate			DATETIME NULL,
+	SubmittedById			UNIQUEIDENTIFIER NULL
+)
+
+--SET THE START AND END DATE
+
+DECLARE @QuarterStartDate date
+DECLARE @QuarterEndDate date
+DECLARE @StartYear int
+DECLARE @EndYear int
+
+	SELECT @StartYear = @ComplianceYear + AddStartYears, @EndYear = @ComplianceYear + AddEndYears FROM [Lookup].[QuarterWindowTemplate] WHERE [Quarter] = @Quarter
+
+	SELECT @QuarterStartDate = DATEFROMPARTS(@StartYear,StartMonth,StartDay), 
+	   @QuarterEndDate = DATEFROMPARTS(@EndYear,EndMonth,EndDay)
+	   FROM [Lookup].[QuarterWindowTemplate] WHERE [Quarter] = @Quarter
+
+	INSERT INTO @AATF
```

Comment:
I think this part of the query for submitted returns will need to use the ReturnAatf table as it's possible the AATF's approval date can change once the the return has been submitted, but this table won't change.
I.e. a created return could have different aatfs to the submitted.

Also can we rename the stored procedure, I don't  like the naming convention :) Can you just rename to getAatfAeReturnDataCsvData

Reply:
Done